ffi/src/ffi_tracing.rs

```diff
@@ -267,25 +270,128 @@ where
                 line: metadata.line().unwrap_or(0),
                 file: kernel_string_slice!(file),
             };
-            (self.callback)(event);
+            let cb = self.callback.lock().unwrap();
```

Comment:
We should not call `unwrap` as it panics and can crash the engine. This is a gross situation because we can't really inform the engine that something went wrong :)  Maybe we can just `error!` log in that case and hope that something else on the rust side is logging.